maintainers/release-process.md

```diff
@@ -0,0 +1,170 @@
+# Nix release process
+
+## Release artifacts
+
+The release process is intended to create the following for each
+release:
+
+* A Git tag
+
+* Binary tarballs in https://releases.nixos.org/?prefix=nix/
+
+* Docker images
+
+* Closures in https://cache.nixos.org
+
+* (Optionally) Updated `fallback-paths.nix` in Nixpkgs
+
+* Update the manual on https://nixos.org/manual/nix/stable/
+
+## Creating a new release from master
+
+* Make sure that https://hydra.nixos.org/jobset/nix/master is green.
+
+* In a checkout of the Nix repo, make sure you're on `master` and do a
+  `git pull`.
+
+* Move the contents of `doc/manual/src/release-notes/rl-next.md`
+  (except the first line) to
+  `doc/manual/src/release-notes/rl-$VERSION.md` (where `$VERSION` is
+  the contents of `.version` *without* the patch level, e.g. `2.12`
+  rather than `2.12.0`).
+
+* Add a header to `doc/manual/src/release-notes/rl-$VERSION.md` like
+
+  ```
+  # Release 2.12 (2022-12-06)
+  ```
+
+* Proof-read / edit / rearrange the release notes. Breaking changes
+  and highlights should go to the top.
+
+* Add a link to the release notes to `doc/manual/src/SUMMARY.md.in`
+  (*not* `SUMMARY.md`), e.g.
+
+  ```
+  - [Release 2.12 (2022-12-06)](release-notes/rl-2.12.md)
+  ```
+
+* Run
+
+  ```console
+  $ git checkout -b release-notes
+  $ git add doc/manual/src/release-notes/rl-$VERSION.md
+  $ git commit -a -m 'Release notes'
+  $ git push --set-upstream edolstra release-notes
+  ```
+
+* Create a PR for `release-notes` and auto-merge it.
+
+* Wait for the PR to be merged.
+
+* Branch the release:
+
+  ```console
+  $ git checkout master
+  $ git pull
+  $ git checkout -b $VERSION-maintenance
+  ```
+
+* Mark the release as stable:
+
+  ```console
+  $ git cherry-pick f673551e71942a52b6d7ae66af8b67140904a76a
```

Comment:
Could we have a tag for this commit so that we don't need to hardcode a commit hash here?

Reply:
This commit only removes a single line from the docs, and usually leads to a merge conflict.

We've discussed automatically removing the line with `sed` in the docs build, when `officialRelease` is true.

Reply:
Yeah that would be better. I don't really want to have a tag for what is just a convenience shortcut...